administrator/components/com_templates/Model/TemplateModel.php

```diff
@@ -148,6 +154,152 @@ public function getDirectoryTree($dir)
+
+		// Only allow html/ folder
+		if($explodeArray['1'] !== 'html')
+		{
+			$app->enqueueMessage(Text::_('COM_TEMPLATES_ERROR_ONLY_HTML_FOLDER_ALLOWED'), 'error');
+
+			return false;
+		}
+
+		$fileName = end($explodeArray);
+		$type     = $explodeArray['2'];
+		$client   = ApplicationHelper::getClientInfo($client_id);
+
+		$componentPath = Path::clean($client->path . '/components/');
+		$modulePath    = Path::clean($client->path . '/modules/');
+		$layoutPath    = Path::clean(JPATH_ROOT . '/layouts/');
+
+		// For modules
+		if (stristr($type, 'mod_') != false)
+		{
+			$folder   = $explodeArray['2'];
+			$htmlPath = Path::clean($modulePath . $folder . '/tmpl/');
+			$fileName = $this->getSafeName($fileName);
+			$coreFile = $this->findPath($htmlPath, $fileName);
+
+			return $coreFile;
+		}
+		elseif (stristr($type, 'com_') != false)
+		{
+			// For components
+			$folder    = $explodeArray['2'];
+			$subFolder = $explodeArray['3'];
+			$fileName  = $this->getSafeName($fileName);
+
+			// The old scheme, if a view has a tmpl folder
+			$oldHtmlPath = Path::clean($componentPath . $folder . '/views/' . $subFolder . '/tmpl/');
+
+			if(!$coreFile = Path::find($oldHtmlPath, $fileName))
+			{
+				// The new scheme, the views are directly in the component/tmpl folder
+				$newHtmlPath = Path::clean($componentPath . $folder . '/tmpl/' . $subFolder . '/');
+				$coreFile    = $this->findPath($newHtmlPath, $fileName);
+
+				return $coreFile;
+			}
+
+			return $coreFile;
+		}
+		elseif (stristr($type, 'layouts') != false)
+		{
+			// For Jlayouts
+			$subtype = $explodeArray['3'];
+
+			if(stristr($subtype, 'com_'))
+			{
+				$folder    = $explodeArray['3'];
+		 		$subFolder = array_slice($explodeArray, 4, -1);
+				$subFolder = implode(DIRECTORY_SEPARATOR, $subFolder);
+				$htmlPath  = Path::clean($componentPath . $folder . '/layouts/' . $subFolder);
+				$fileName  = $this->getSafeName($fileName);
+				$coreFile  = $this->findPath($htmlPath, $fileName);
+
+				return $coreFile;
+			}
+			elseif (stristr($subtype, 'joomla') || stristr($subtype, 'libraries') || stristr($subtype, 'plugins'))
+			{
+				$subFolder = array_slice($explodeArray, 3, -1);
+				$subFolder = implode(DIRECTORY_SEPARATOR, $subFolder);
+				$htmlPath  = Path::clean($layoutPath . $subFolder);
+				$fileName  = $this->getSafeName($fileName);
+				$coreFile  = $this->findPath($htmlPath, $fileName);
+
+				return $coreFile;
+			}
+		}
+		
+		$app->enqueueMessage(Text::_('COM_TEMPLATES_ERROR_CORE_FILE_NOT_FOUND'), 'error');
+
+		return false;
+	}
+
+	/**
+	 * Creates a safe file name for the given name.
+	 *
+	 * @param   string  $name  The filename
+	 *
+	 * @return  string $fileName  The filtered name without Date
+	 *
+	 * @since   4.0.0
+	 */
+	private function getSafeName($name)
+	{
+		if (preg_match('/[0-9]/', $name))
+		{
+			// Get the extension
+			$extension = File::getExt($name);
+
+			// Remove ( Date ) from file
+			$explodeArray         = explode('-', $name);
```

Comment:
How do we get a date into the file name?

Reply:
![selection_014](https://user-images.githubusercontent.com/30978328/40361461-096b9f1c-5de7-11e8-9abf-977c2abac9c3.png)

Try to create one override two times, then it add unique name by adding current Date to it .

Reply:
Didn't know that!